src/pybind/mgr/cephadm/module.py

```diff
@@ -1665,15 +1665,49 @@ def _get_hosts(self, label=None):
                 r.append(h)
         return r
 
+    def _verify_add_host(self, spec: HostSpec):
+        """
+        Make sure, we have sane hostname.
+        """
+        message = """
+  You may add `--force` to ignore this. Be aware that this might
+  cause CEPHADM_STRAY_HOST HEALTH warnings."""
+        existing_ceph: Set[str] = {s['hostname'] for s in self.list_servers() if
+                                   'hostname' in s}
+
+        h = spec.hostname
+        for other in existing_ceph:
+            if h.startswith(other + '.') or other.startswith(h + '.'):
+                # TODO: under some artificial circumstances, this makes it impossible to
+                # add hosts, e.g. add_host('myhost') && add_host('myhost.with-a-dot-in-it')
+                raise OrchestratorValidationError(
+                    f"Failed to add {h}: There is already a similar hostname: {other}" + message)
+
+        if self.inventory.keys():
+            fqdns = [i.count('.') > 0 for i in self.inventory.keys()]
+            any_fqds = any(fqdns)
+            if h.count('.') > 0 and not any_fqds:
```

Comment:
I think `any()` and `all()` are pretty understandable:
```suggestion
            if h.count('.') > 0 and not any(fqdns):
```